scaleway/resource_instance_server.go

```diff
@@ -231,38 +231,38 @@ func resourceScalewayInstanceServerCreate(d *schema.ResourceData, m interface{})
 
 	commercialType := d.Get("type").(string)
 
-	image := expandID(d.Get("image"))
-	if !isUUID(image) {
+	image := expandZonedID(d.Get("image"))
+	if !isUUID(image.ID) {
 		instanceAPI := marketplace.NewAPI(m.(*Meta).scwClient)
 		imageUUID, err := instanceAPI.GetLocalImageIDByLabel(&marketplace.GetLocalImageIDByLabelRequest{
 			CommercialType: commercialType,
 			Zone:           zone,
-			ImageLabel:     image,
+			ImageLabel:     image.ID,
 		})
 		if err != nil {
 			return fmt.Errorf("could not get image '%s': %s", image, err)
 		}
-		image = imageUUID
+		image = expandZonedID(imageUUID)
```

Comment:
expand should be reserved for when you read from `state` . In this case we should have something like
```
image = newZonedId(imageUUID, zone)
```